### PR TITLE
enableRbac added in data_fusion_instance along with test

### PR DESCRIPTION
**Pull request by sanghaniJ**
Added Support for enable_rbac in data_fusion_instance
fixes https://github.com/hashicorp/terraform-provider-google/issues/9065

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
datafusion: added `enable_rbac` field to `google_data_fusion_instance` resource
```

### CODE REVIEW

File: mmv1/products/datafusion/api.yaml [hunk 1]
```diff
@@ -103,6 +103,10 @@ objects:
         name: 'enableStackdriverMonitoring'
         description: |
           Option to enable Stackdriver Monitoring.
+      - !ruby/object:Api::Type::Boolean
```
Comment (shuyama1) [inline mmv1/products/datafusion/api.yaml:106]: Do we know if the field can be updated in place? If not, we need to add `input: true`. More details can be found at https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1#apiyaml

Comment (sanghaniJ) [reply mmv1/products/datafusion/api.yaml:106]: The behavior of this field is quite different. Hence, will need to have a look and check if that is the case, if so I'll add the necessary changes.

Comment (sanghaniJ) [reply mmv1/products/datafusion/api.yaml:106]: I've tried to update the field and the apply passes but the value doesn't get updated. So, the patch call passes the field with new value and in the UI it also says Updating but when we check the value returned (state file), it is unchanged. I've discussed this with Stephen and he suggested to take your inputs on this. Any thoughts?

Comment (shuyama1) [reply mmv1/products/datafusion/api.yaml:106]: yeah, if you tested it and confirmed that the field was't able to be updated, then adding `input: true` sounds good.

Comment (sanghaniJ) [reply mmv1/products/datafusion/api.yaml:106]: Thanks, I've added `input:true` to this field.

**Review by shuyama1 — CHANGES_REQUESTED** [review]
Based on the [API documentation](https://cloud.google.com/data-fusion/docs/reference/rest/v1beta1/projects.locations.instances/patch#query-parameters), it looks like that `updateMask` is used to update fields in this resource. Therefore, can we try adding `update_mask: true` to the resource in the api.yaml file (something like https://github.com/GoogleCloudPlatform/magic-modules/blob/af4dbeaeebd191c5dd5bfe3f782866c9ac11d7d1/mmv1/products/identityplatform/api.yaml#L36) and removing `input: true` of the field `enableRbac`. 

Plus, can we add an update test for this field? Currently, all update tests are handwritten, and it can be added at [resource_data_fusion_instance_test.go.erb file](https://github.com/GoogleCloudPlatform/magic-modules/mmv1/third_party/terraform/tests/resource_data_fusion_instance_test.go.erb). 

Let me know if you need help implementing it! Thanks!

**Comment by sanghaniJ** [comment]
@shuyama1 For testing, is there anything which gets affected due to update_mask ? I couldn't find any documentation for that and I've a failing test which says that I cannot add field labels in the update test and there are only certain values which can be updated.

**Review by shuyama1 — COMMENTED** [review]
@sanghaniJ Looks like the API doc says `A field will be overwritten if it is in the mask. If the user does not provide a mask, all the supported fields (labels and options currently) will be overwritten.` It sounds like that `labels` and `options` do not need to be included in the `updatemask`. 

I only tried the changes in the downstream providers -- commented out the lines
```
if d.HasChange("labels") {
		updateMask = append(updateMask, "labels")
}
```
in `resourceDataFusionInstanceUpdate` and it seems that the update tests passed. However, since it is a generated resource, we may need custom `pre_update` for the update function. Some `pre_update` examples can be found at https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/pre_update

**Comment by sanghaniJ** [comment]
@shuyama1 I've seen a similar case in resource [active_directory_domain](https://github.com/GoogleCloudPlatform/magic-modules/blob/f67f3966a9dc60a18fd02b199a4999770ba398bd/mmv1/products/activedirectory/api.yaml#L103 ) where update_mask is added and labels field gets updated in [active_directory_domain_update_test](https://github.com/GoogleCloudPlatform/magic-modules/blob/f67f3966a9dc60a18fd02b199a4999770ba398bd/mmv1/third_party/terraform/tests/resource_active_directory_domain_update_test.go#L73) while the same fails for us, is there any reason behind this ?

**Comment by shuyama1** [comment]
I guess it depends on the design of the API. For `active_directory_domain` resource, it looks like the `labels` field is accepted in the updateMask based on [the API doc](https://cloud.google.com/managed-microsoft-ad/reference/rest/v1beta1/projects.locations.global.domains/patch#query-parameters), while for this resource, it seems that the `labels` field is not (?) based on [the doc](https://cloud.google.com/data-fusion/docs/reference/rest/v1beta1/projects.locations.instances/patch#query-parameters).
But it's definitely worth checking with the service team if `labels` and `options` should be excluded from the updateMask.

**Comment by sanghaniJ** [comment]
Can you check with the service team on my behalf ? As I currently don't have authority to do so.

**Comment by sanghaniJ** [comment]
@shuyama1 I've added  the update mask with fields enableRbac, enableStackdriverMonitoring and enableStackdriverLogging using the pre_update function and it passes all the checks as well.

**Review by shuyama1 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**